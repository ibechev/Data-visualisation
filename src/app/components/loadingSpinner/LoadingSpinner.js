import React from "react";
import PropTypes from "prop-types";

const LoadingSpinner = ({ title }) => {
  return (
    <div className="loading-spinner">
      <i className="fas fa-spinner fa-spin" />
      <p className="loading-title">Loading {title.toLowerCase()} ...</p>
    </div>
  );
};

LoadingSpinner.propTypes = {
  title: PropTypes.string.isRequired
};

export default LoadingSpinner;
