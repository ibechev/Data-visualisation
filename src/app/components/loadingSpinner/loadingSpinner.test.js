import React from "react";
import { shallow } from "enzyme";
import LoadingSpinner from "./LoadingSpinner";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../tests/config/testUtils";

describe("Expect <LoadingSpinner />", () => {
  it("to render correctly", () => {
    const wrapper = renderer.create(<LoadingSpinner title="title" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with corrent props", () => {
    propTypeErrors(<LoadingSpinner title="title" />, { title: "title" });
  });

  it("to display prop 'title' in <p class='loading-title'", () => {
    const title = "title";
    const wrapper = shallow(<LoadingSpinner title={title} />);
    expect(wrapper.find("p.loading-title").text()).toEqual(
      `Loading ${title} ...`
    );
  });
});
