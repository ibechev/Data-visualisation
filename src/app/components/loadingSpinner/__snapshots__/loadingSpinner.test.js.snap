// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LoadingSpinner /> to render correctly 1`] = `
<div
  className="loading-spinner"
>
  <i
    className="fas fa-spinner fa-spin"
  />
  <p
    className="loading-title"
  >
    Loading 
    title
     ...
  </p>
</div>
`;
